src/synapse-plugins/calculator-plugin/calculator-plugin-backend.vala

```diff
@@ -115,5 +101,42 @@ namespace Synapse {
                 return double.parse (solution);
             }
         }
+
+
+        private bool allowed_expression (string expr, out bool is_base_expr) {
+            is_base_expr = false;
+            if (expr.length > 6 &&
+                (expr.has_prefix ("ibase=") || expr.has_prefix ("obase="))
```

Comment:
We should probably have some unit tests for this ibase and obase functionality?

Reply:
Tests added